src/main/java/fr/insalyon/creatis/agent/workflow/command/KillCommand.java

```diff
@@ -15,24 +17,33 @@ public KillCommand(JobDatabase jobDB) {
     }
 
     public void run() {
-        List<String> jobIds = this.jobDB.getJobIds();
-
-        for (String id : jobIds) {
-            try {
-                logger.info("Killing job id '" + id + "'");
-                String exec = "dirac-wms-job-kill " + id;
-                Process process = Runtime.getRuntime().exec(exec);
-                process.waitFor();
-                if (process.exitValue() != 0) {
-                    logger.error("Unable to kill job id '" + id + "'");
-                } else {
-                    this.jobDB.updateStatus(id);
-                }
-            } catch (InterruptedException | IOException e) {
-                logger.error("Error killing job " + id, e);
-            }
+        Map<String,String> jobIds = this.jobDB.getJobsToKill();
+
+        for (String id : jobIds.keySet()) {
+            // kill running jobs, delete waiting and succesfully submitted
+            boolean mustDelete = ! jobIds.get(id).equalsIgnoreCase("RUNNING");
+            killJob(id, mustDelete);
         }
 
         this.jobDB.close();
     }
+
+    public void killJob(String id, boolean mustDelete) {
+        try {
+            String command = mustDelete ? "delete" : "kill";
```

Comment:
We agree that we only make this distinction so that we keep a maximum of traces for running jobs, right? From a functional point of view, we could call delete on all of them.